pennylane/pulse/rydberg_hamiltonian.py

```diff
@@ -122,13 +122,13 @@ def circuit():
     )
 
 
-def rydberg_drive(amplitude, phase, detuning, wires):
+def rydberg_drive(amplitude, phase, wires, detuning=None):
```

Comment:
I hadn't considered argument order as an implication of adding a default value for detuning. 

I like that it doesn't create the detuning term with a coefficient of 0, but I'm not sure about separating out detuning from the other parameters so that it can default to None. Maybe users just need to pass 0 to detuning if they want to drive on resonance.

What do other people think?

Reply:
I don't mind that change. However, I think that there is some benefit to having `detuning` be a kwarg, as that explicitly means that it is an _optional_ parameter (that is, if we want it to be interpreted as one).

Reply:
If it's about optional parameters, I would also make `phase` a kwarg as well, as that would also be optional. I would say `phase` isn't more or less optional than `detuning` in terms of defining a pulse: the simplest drive would be on-resonance with no phase shift, i.e. `phase=0`, and `detuning=0`, and in many instances you would drive off-resonance without adding any phase shift.

Reply:
And honestly since it's also for adding local shift drive, `amplitude` is also optional. Maybe this should have `wires` as the first argument, and then everything else is kwargs? But I guess the convention in PL is usually first parameters, then wires.

Or just have all of them be required parameters, and force people to always make a decision about all the attributes of the generated pulse.

Reply:
I think if everything can be a kwarg, then probably nothing should be a kwarg 😅 . And thanks for reminding me about PL convention. Generally trainable parameters are positional arguments. This function is by no means conventional in many different ways, but it seems like making everything positional arguments (as it was before this PR) might be the way to go. Let me know what everyone thinks and I can revert `detuning` back to a positional argument if we decide to do so.

Reply:
I was reading through this thread and thought "if they are optional, kwargs sound good", but the argument that we have trainable parameters be positional arguments seems to be the stronger one for me personally as well.
I'm wondering whether there would be user-facing convenience functions that alias back to `rydberg_drive` with certain parameters set to 0? :thinking:

Reply:
Note that those parameters are not the trainable parameters themselves, but callables with trainable parameters.

If one wants to make the _constants_ trainable, the "correct" (/advertised) way is to pass constant callables (that are later called through the Hamiltonian's `params`).

Reply:
Personally don't have a strong opinion here and wouldnt mind leaving it as is

Reply:
> I would also make phase a kwarg as well
👍